### ISSUE TITLE
Rename Current Workspace

### ISSUE THREAD

**Issue by danielnbalasoiu** [wez/wezterm#2787]
According to [the documentation](https://wezfurlong.org/wezterm/config/lua/keyassignment/SwitchToWorkspace.html), you can switch to an existing workspace or create a new one if it doesn't exist.
If `name` parameter is omitted, a randomly generated name will be chosen. Once the workspace is created, there is no option to rename it.

Please take this into consideration and include a feature to rename the current workspace.

Thank you!

P.S.: I ❤️ Wezterm! You 🤘 @wez!

**Comment by aniketgm** [comment]
Is anyone working on this ?
There are two ways AFAICT (as far as I can think),
1. Either provide an option in the LauncherMenu to rename the current workspace,
When the option is clicked/enter is pressed on selection, a rename prompt will be appear at the bottom to rename the current workspace, like currently there is for QuickSelect option.
2. Provide the rename feature as a function that one can create their own keymaps out of it and launch it with their own bindings like for e.g. `wezterm.action.RenameCurrentWorkspace` something like this.
@wez @danielnbalasoiu what'd you think ?

**Comment by wez** [comment]
No one is working on this.
There isn't currently a convenient UI for prompting the user for text input that can be easily coupled to an action.

It is possible to specify the name of a workspace at creation time if you define a key assignment or use the [gui-startup](https://wezfurlong.org/wezterm/config/lua/gui-events/gui-startup.html) to pre-define them.

> I feel like the random naming thing was too much of a feature added. I mean simply naming them with numbers would've sufficed. 

The tone of what you've written comes across as: "I feel that this isn't the way it should be, why isn't someone working to build it the way I want it?", which is very entitled.  There's one of me and thousands of users.  There are a lot of unfinished features because I have limited time to spend on them and I do this for fun.

When someone makes entitled demands, I actively ignore the associated feature requests because I do not want to enable that behavior.

**Comment by aniketgm** [comment]
> The tone of what you've written comes across as: "I feel that this isn't the way it should be, why isn't someone working to build it the way I want it?", which is very entitled.

I apologize if that sentence came out wrong. Please don't take it personally. Or rather just ignore that statement. You have created an amazing tool. I just wanted to convey what I felt could've been a simple design. Sometimes we overthink things when trying to create something good, which wezterm indeed is. My main point was the last sentence.

> It is possible to specify the name of a workspace at creation time if you define a key assignment or use the [gui-startup](https://wezfurlong.org/wezterm/config/lua/gui-events/gui-startup.html) to pre-define them.

Yup! currently, the workaround I have is with gui-startup. I'm not sure how to do it with a key assignment. I'll check it out.

**Comment by wez** [comment]
In `main` and the latest nightly builds:

* [wezterm cli rename-workspace](https://wezfurlong.org/wezterm/cli/cli/rename-workspace.html)
* [wezterm.mux.rename_workspace()](https://wezfurlong.org/wezterm/config/lua/wezterm.mux/rename_workspace.html)

**Comment by danielnbalasoiu** [comment]
Just tested `wezterm cli rename-workspace` and it worked. 🥳 

Really really appreciat it! 🙏 🚀

**Comment by torypages** [comment]
This is amazing. This weekend I hoped to transition to Wezterm to be ready Monday morning. I hit roadblocks with tab and workspace rename. I saw the tab rename commit come in, and for the last little bit I've been hacking in a way to make a workspace with a name. It was seriously bad and embarrassing (would have never made it a PR), but it was probably going to work. But now I see this!!! And just tested it!

I think I've got everything I need to make this usable now. This timing is so amazing it's creepy. I'm so excited to bring my ssh+tmux experience to something that feels more local and natural.

As a Rust beginner it was fun poking around the code, I hope to do some more.

Thank you!
